### ISSUE TITLE
Fail to install agent on windows 2012r2-2016 from a puppet master

### ISSUE THREAD

**Issue by fduranti** [oscar-stack/vagrant-pe_build#139]
It seems I've a problem provisioning puppet_agent with the vagrant-pe-build plugin.
I've correctly provisioned the puppet master on a centos machine.
When I try to install/provision a windows 2012R2/2016 machine the puppet agent is not installed.
In the c:\Users\vagrant\Documents directory there's only the installation log but not the puppet-agent msi package.
It seems that the package is not downloaded from the server and when the installation get run it get a 1639 error.

This is the output of vagrant up:

```
==> win2016pmd15.q8inttest.com: Running provisioner: pe_agent...
==> pmaster.q8inttest.com: The Puppet Server on pmaster.q8inttest.com is configured with Agent repositories for:
==> pmaster.q8inttest.com:   windows-x86_64
==> win2016pmd15.q8inttest.com: Downloading PE installer from:
==> win2016pmd15.q8inttest.com:   https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi
==> win2016pmd15.q8inttest.com: Running: $params = @(
==> win2016pmd15.q8inttest.com:   "/qn",
==> win2016pmd15.q8inttest.com:   "/i `"puppet-agent-x64.msi`"",
==> win2016pmd15.q8inttest.com:   "/l*v puppet-enterprise-installer.log",
==> win2016pmd15.q8inttest.com:   "PUPPET_MASTER_SERVER=pmaster PUPPET_AGENT_CERTNAME=win2016pmd15.q8inttest.com"
==> win2016pmd15.q8inttest.com: )
==> win2016pmd15.q8inttest.com: 
==> win2016pmd15.q8inttest.com: (Start-Process -FilePath "msiexec.exe" -ArgumentList $params -Wait -Passthru).ExitCode
==> win2016pmd15.q8inttest.com: 1619
==> pmaster.q8inttest.com: No pending signing request for  on pmaster.q8inttest.com. Skipping autosign.
```

This is part of the debug log during the download and it seems that it checks the certificate of the server:
```
 INFO guest: Execute capability: stage_installer [#<Vagrant::Machine: win2016pmd15.q8inttest.com (VagrantPlugins::ProviderVirtualBox::Provider)>, #<URI::HTTPS https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi>, "."] (windows)
DEBUG winrmshell: [WinRM] opening remote shell on http://127.0.0.1:5985/wsman
DEBUG winrmshell: [WinRM] Waiting for output...
DEBUG winrmshell: [WinRM] Processing output
DEBUG winrmshell: [WinRM] polling for pipeline state. message: #<WinRM::PSRP::Message:0x00007fc8314f0dc0 @data="\xEF\xBB\xBF<Obj RefId=\"0\"><MS><Version N=\"protocolversion\">2.3</Version><Version N=\"PSVersion\">2.0</Version><Version N=\"SerializationVersion\">1.1.0.1</Version></MS></Obj>", @destination=1, @type=65538, @pipeline_id="00000000-0000-0000-0000-000000000000", @runspace_pool_id="00000000-0000-0000-0000-000000000000">
DEBUG winrmshell: [WinRM] polling for pipeline state. message: #<WinRM::PSRP::Message:0x00007fc8314efe98 @data="\xEF\xBB\xBF<Obj RefId=\"0\"><MS><Obj N=\"ApplicationPrivateData\" RefId=\"1\"><TN RefId=\"0\"><T>System.Management.Automation.PSPrimitiveDictionary</T><T>System.Collections.Hashtable</T><T>System.Object</T></TN><DCT><En><S N=\"Key\">BreakAll</S><B N=\"Value\">false</B></En><En><S N=\"Key\">DebugStop</S><B N=\"Value\">false</B></En><En><S N=\"Key\">UnhandledBreakpointMode</S><I32 N=\"Value\">1</I32></En><En><S N=\"Key\">PSVersionTable</S><Obj N=\"Value\" RefId=\"2\"><TNRef RefId=\"0\" /><DCT><En><S N=\"Key\">PSVersion</S><Version N=\"Value\">5.1.14393.206</Version></En><En><S N=\"Key\">PSEdition</S><S N=\"Value\">Desktop</S></En><En><S N=\"Key\">PSCompatibleVersions</S><Obj N=\"Value\" RefId=\"3\"><TN RefId=\"1\"><T>System.Version[]</T><T>System.Array</T><T>System.Object</T></TN><LST><Version>1.0</Version><Version>2.0</Version><Version>3.0</Version><Version>4.0</Version><Version>5.0</Version><Version>5.1.14393.206</Version></LST></Obj></En><En><S N=\"Key\">CLRVersion</S><Version N=\"Value\">4.0.30319.42000</Version></En><En><S N=\"Key\">BuildVersion</S><Version N=\"Value\">10.0.14393.206</Version></En><En><S N=\"Key\">WSManStackVersion</S><Version N=\"Value\">3.0</Version></En><En><S N=\"Key\">PSRemotingProtocolVersion</S><Version N=\"Value\">2.3</Version></En><En><S N=\"Key\">SerializationVersion</S><Version N=\"Value\">1.1.0.1</Version></En></DCT></Obj></En><En><S N=\"Key\">DebugMode</S><I32 N=\"Value\">1</I32></En><En><S N=\"Key\">DebugBreakpointCount</S><I32 N=\"Value\">0</I32></En></DCT></Obj></MS></Obj>", @destination=1, @type=135177, @pipeline_id="00000000-0000-0000-0000-000000000000", @runspace_pool_id="00000000-0000-0000-0000-000000000000">
DEBUG winrmshell: [WinRM] Waiting for output...
DEBUG winrmshell: [WinRM] Processing output
DEBUG winrmshell: [WinRM] polling for pipeline state. message: #<WinRM::PSRP::Message:0x00007fc83149e1b0 @data="\xEF\xBB\xBF<Obj RefId=\"0\"><MS><I32 N=\"RunspaceState\">2</I32></MS></Obj>", @destination=1, @type=135173, @pipeline_id="00000000-0000-0000-0000-000000000000", @runspace_pool_id="00000000-0000-0000-0000-000000000000">
DEBUG winrmshell: [WinRM] remote shell created with shell_id: 3E270A19-07A3-47A9-AAC3-C9DE75A76AB4
DEBUG winrmshell: [WinRM] Command created for If (Test-Path "./puppet-agent-x64.msi) { Exit 0 } Else { Exit 1 }
if (!$?) { if($LASTEXITCODE) { exit $LASTEXITCODE } else { exit 1 } } with id: 968A1F49-28EF-4960-97E0-F6BAFB5D71F2
DEBUG winrmshell: [WinRM] creating command_id: 968A1F49-28EF-4960-97E0-F6BAFB5D71F2 on shell_id 3E270A19-07A3-47A9-AAC3-C9DE75A76AB4
DEBUG winrmshell: [WinRM] Waiting for output...
DEBUG winrmshell: [WinRM] Processing output
DEBUG winrmshell: [WinRM] cleaning up command_id: 968A1F49-28EF-4960-97E0-F6BAFB5D71F2 on shell_id 3E270A19-07A3-47A9-AAC3-C9DE75A76AB4
DEBUG winrmshell: Output: #<WinRM::Output:0x00007fc83139ecd8 @data=[{:stderr=>"At line:1 char:15\r\n+ If (Test-Path \"./puppet-agent-x64.msi) { Exit 0 } Else { Exit 1 }\r\n+               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\nThe string is missing the terminator: \".\r\n\r\nAt line:2 char:71\r\n+ if (!$?) { if($LASTEXITCODE) { exit $LASTEXITCODE } else { exit 1 } }\r\n+                                                                       ~\nMissing closing ')' after expression in 'If' statement.\n    + CategoryInfo          : ParserError: (:) [Invoke-Expression], ParseException\n    + FullyQualifiedErrorId : TerminatorExpectedAtEndOfString,Microsoft.PowerShell.Commands.InvokeExpressionCommand\n"}], @exitcode=0>
 WARN winrmshell: Detected ParserError, setting exit code to 1
 INFO interface: info: Downloading PE installer from:
  https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi
 INFO interface: info: ==> win2016pmd15.q8inttest.com: Downloading PE installer from:
==> win2016pmd15.q8inttest.com:   https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi
==> win2016pmd15.q8inttest.com: Downloading PE installer from:
==> win2016pmd15.q8inttest.com:   https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi
DEBUG winrm: powershell executing:
[System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
(New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi","./puppet-agent-x64.msi")

DEBUG winrmshell: [WinRM] opening remote shell on http://127.0.0.1:5985/wsman
DEBUG winrmshell: [WinRM] Waiting for output...
DEBUG winrmshell: [WinRM] Processing output
DEBUG winrmshell: [WinRM] polling for pipeline state. message: #<WinRM::PSRP::Message:0x00007fc8307f2998 @data="\xEF\xBB\xBF<Obj RefId=\"0\"><MS><Version N=\"protocolversion\">2.3</Version><Version N=\"PSVersion\">2.0</Version><Version N=\"SerializationVersion\">1.1.0.1</Version></MS></Obj>", @destination=1, @type=65538, @pipeline_id="00000000-0000-0000-0000-000000000000", @runspace_pool_id="00000000-0000-0000-0000-000000000000">
DEBUG winrmshell: [WinRM] polling for pipeline state. message: #<WinRM::PSRP::Message:0x00007fc8307e3970 @data="\xEF\xBB\xBF<Obj RefId=\"0\"><MS><Obj N=\"ApplicationPrivateData\" RefId=\"1\"><TN RefId=\"0\"><T>System.Management.Automation.PSPrimitiveDictionary</T><T>System.Collections.Hashtable</T><T>System.Object</T></TN><DCT><En><S N=\"Key\">BreakAll</S><B N=\"Value\">false</B></En><En><S N=\"Key\">DebugStop</S><B N=\"Value\">false</B></En><En><S N=\"Key\">UnhandledBreakpointMode</S><I32 N=\"Value\">1</I32></En><En><S N=\"Key\">PSVersionTable</S><Obj N=\"Value\" RefId=\"2\"><TNRef RefId=\"0\" /><DCT><En><S N=\"Key\">PSVersion</S><Version N=\"Value\">5.1.14393.206</Version></En><En><S N=\"Key\">PSEdition</S><S N=\"Value\">Desktop</S></En><En><S N=\"Key\">PSCompatibleVersions</S><Obj N=\"Value\" RefId=\"3\"><TN RefId=\"1\"><T>System.Version[]</T><T>System.Array</T><T>System.Object</T></TN><LST><Version>1.0</Version><Version>2.0</Version><Version>3.0</Version><Version>4.0</Version><Version>5.0</Version><Version>5.1.14393.206</Version></LST></Obj></En><En><S N=\"Key\">CLRVersion</S><Version N=\"Value\">4.0.30319.42000</Version></En><En><S N=\"Key\">BuildVersion</S><Version N=\"Value\">10.0.14393.206</Version></En><En><S N=\"Key\">WSManStackVersion</S><Version N=\"Value\">3.0</Version></En><En><S N=\"Key\">PSRemotingProtocolVersion</S><Version N=\"Value\">2.3</Version></En><En><S N=\"Key\">SerializationVersion</S><Version N=\"Value\">1.1.0.1</Version></En></DCT></Obj></En><En><S N=\"Key\">DebugMode</S><I32 N=\"Value\">1</I32></En><En><S N=\"Key\">DebugBreakpointCount</S><I32 N=\"Value\">0</I32></En></DCT></Obj></MS></Obj>", @destination=1, @type=135177, @pipeline_id="00000000-0000-0000-0000-000000000000", @runspace_pool_id="00000000-0000-0000-0000-000000000000">
DEBUG winrmshell: [WinRM] Waiting for output...
DEBUG winrmshell: [WinRM] Processing output
DEBUG winrmshell: [WinRM] polling for pipeline state. message: #<WinRM::PSRP::Message:0x00007fc8305ad278 @data="\xEF\xBB\xBF<Obj RefId=\"0\"><MS><I32 N=\"RunspaceState\">2</I32></MS></Obj>", @destination=1, @type=135173, @pipeline_id="00000000-0000-0000-0000-000000000000", @runspace_pool_id="00000000-0000-0000-0000-000000000000">
DEBUG winrmshell: [WinRM] remote shell created with shell_id: 0C6912A2-8AA6-4E67-BF0B-D4A54934A5A5
DEBUG winrmshell: [WinRM] Command created for [System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
(New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi","./puppet-agent-x64.msi")

if (!$?) { if($LASTEXITCODE) { exit $LASTEXITCODE } else { exit 1 } } with id: 7478CEF7-1A94-4907-A6BA-9882EAF5C416
DEBUG winrmshell: [WinRM] creating command_id: 7478CEF7-1A94-4907-A6BA-9882EAF5C416 on shell_id 0C6912A2-8AA6-4E67-BF0B-D4A54934A5A5
DEBUG winrmshell: [WinRM] Waiting for output...
DEBUG winrmshell: [WinRM] Processing output
DEBUG winrmshell: [WinRM] cleaning up command_id: 7478CEF7-1A94-4907-A6BA-9882EAF5C416 on shell_id 0C6912A2-8AA6-4E67-BF0B-D4A54934A5A5
DEBUG winrmshell: Output: #<WinRM::Output:0x00007fc83038f860 @data=[], @exitcode=0>

```

Running the command from a powershell i get the certificate error:
```
PS C:\Users\Administrator> (New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/packages/current/window
-x86_64/puppet-agent-x64.msi","./puppet-agent-x64.msi")
Exception calling "DownloadFile" with "2" argument(s): "The underlying connection was closed: Could not establish
trust relationship for the SSL/TLS secure channel."
At line:1 char:1
+ (New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/ ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
```

THis is the puppet-installer.log:
```
=== Verbose logging started: 5/8/2018  4:40:22  Build type: SHIP UNICODE 5.00.10011.00  Calling process: C:\Windows\system32\msiexec.exe ===
MSI (c) (C4:D0) [04:40:22:743]: Resetting cached policy values
MSI (c) (C4:D0) [04:40:22:743]: Machine policy value 'Debug' is 0
MSI (c) (C4:D0) [04:40:22:743]: ******* RunEngine:
           ******* Product: puppet-agent-x64.msi
           ******* Action: 
           ******* CommandLine: **********
MSI (c) (C4:D0) [04:40:22:758]: Client-side and UI is none or basic: Running entire install on the server.
MSI (c) (C4:D0) [04:40:22:758]: Grabbed execution mutex.
MSI (c) (C4:D0) [04:40:22:821]: Cloaking enabled.
MSI (c) (C4:D0) [04:40:22:821]: Attempting to enable all disabled privileges before calling Install on Server
MSI (c) (C4:D0) [04:40:22:821]: Incrementing counter to disable shutdown. Counter after increment: 0
MSI (s) (6C:14) [04:40:23:007]: Running installation inside multi-package transaction C:\Users\vagrant\Documents\puppet-agent-x64.msi
MSI (s) (6C:14) [04:40:23:007]: Grabbed execution mutex.
MSI (s) (6C:1C) [04:40:23:007]: Resetting cached policy values
MSI (s) (6C:1C) [04:40:23:007]: Machine policy value 'Debug' is 0
MSI (s) (6C:1C) [04:40:23:007]: ******* RunEngine:
           ******* Product: C:\Users\vagrant\Documents\puppet-agent-x64.msi
           ******* Action: 
           ******* CommandLine: **********
MSI (s) (6C:1C) [04:40:23:116]: Note: 1: 2203 2: C:\Users\vagrant\Documents\puppet-agent-x64.msi 3: -2147287038 
MSI (s) (6C:1C) [04:40:23:116]: MainEngineThread is returning 2
MSI (s) (6C:14) [04:40:23:163]: User policy value 'DisableRollback' is 0
MSI (s) (6C:14) [04:40:23:163]: Machine policy value 'DisableRollback' is 0
MSI (s) (6C:14) [04:40:23:163]: Incrementing counter to disable shutdown. Counter after increment: 0
MSI (s) (6C:14) [04:40:23:194]: Note: 1: 1402 2: HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Installer\Rollback\Scripts 3: 2 
MSI (s) (6C:14) [04:40:23:194]: Note: 1: 1402 2: HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Installer\Rollback\Scripts 3: 2 
MSI (s) (6C:14) [04:40:23:210]: Decrementing counter to disable shutdown. If counter >= 0, shutdown will be denied.  Counter after decrement: -1
MSI (c) (C4:D0) [04:40:23:210]: Decrementing counter to disable shutdown. If counter >= 0, shutdown will be denied.  Counter after decrement: -1
MSI (c) (C4:D0) [04:40:23:210]: MainEngineThread is returning 2
=== Verbose logging stopped: 5/8/2018  4:40:23 ===

```

**Comment by Sharpie** [comment]
The line run before `DownloadFile`:

    [System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}

Should have set the cert validation check to a function that just returns `true` and allows the check to pass. The exit code for the download appears to be zero, but the code from msiexec indicates that it had a problem locating the msi file.

Was there a `puppet-agent-x64.msi` sitting next to where the `puppet-installer.log`? If not, do you get a different result from running `DownloadFile` manually if the `ServerCertificateValidationCallback` line is run first?

**Comment by fduranti** [comment]
The plugin fail to download the file puppet-agent-x64.msi and the msiexec return an error code 1619 and nothing is installed.
I see that line but I don't have any control over it, I don't know how it can be changed... the download is done from the vagrant-pe_build module itself. Is there any parameter to change that value?

The error in the installer log is exactly the fact that the file is not found (for the download error).
Manually downloading the file into the c:\users\vagrant\documents directory work and in that case the plugin is able to install puppet correctly.

**Comment by Sharpie** [comment]
Hmm, so running the following in an admin shell results in `puppet-agent-x64.msi` being downloaded?

```
[System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
(New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi","./puppet-agent-x64.msi")
```

**Comment by fduranti** [comment]
This is really strange. I logged on with the vagrant user that is used during vagrant provisioning.
If I  run that command in c:\users\vagrant it work correctly and download the file
If I run that command in the c:\users\vagrant\documents that is the directory in which the puppet-installer.log file is found so where the vagrant home in when downloading the file and trying to install it it will not download it.
```
PS C:\Users\vagrant> [System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
PS C:\Users\vagrant> (New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/packages/current/windows-x86_6
4/puppet-agent-x64.msi","./puppet-agent-x64.msi")
PS C:\Users\vagrant>
PS C:\Users\vagrant> dir


    Directory: C:\Users\vagrant


Mode                LastWriteTime     Length Name
----                -------------     ------ ----
d----          5/8/2018   1:43 PM            .ssh
d-r--          5/8/2018   1:42 PM            Contacts
d-r--          5/8/2018   1:42 PM            Desktop
d-r--          5/8/2018  11:35 AM            Documents
d-r--          5/8/2018   1:42 PM            Downloads
d-r--          5/8/2018   1:42 PM            Favorites
d-r--          5/8/2018   1:42 PM            Links
d-r--          5/8/2018   1:42 PM            Music
d-r--          5/8/2018   1:42 PM            Pictures
d-r--          5/8/2018   1:42 PM            Saved Games
d-r--          5/8/2018   1:42 PM            Searches
d-r--          5/8/2018   1:42 PM            Videos
-a---          5/8/2018   1:42 PM          6 .vbox_version
-a---          5/8/2018  11:38 AM   30942208 puppet-agent-x64.msi


PS C:\Users\vagrant> cd .\Documents
PS C:\Users\vagrant\Documents> dir


    Directory: C:\Users\vagrant\Documents


Mode                LastWriteTime     Length Name
----                -------------     ------ ----
-a---          5/8/2018  11:35 AM       5074 puppet-enterprise-installer.log


PS C:\Users\vagrant\Documents> [System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}
PS C:\Users\vagrant\Documents> (New-Object System.Net.WebClient).DownloadFile("https://pmaster:8140/packages/current/win
dows-x86_64/puppet-agent-x64.msi","./puppet-agent-x64.msi")
PS C:\Users\vagrant\Documents> dir


    Directory: C:\Users\vagrant\Documents


Mode                LastWriteTime     Length Name
----                -------------     ------ ----
-a---          5/8/2018  11:35 AM       5074 puppet-enterprise-installer.log


PS C:\Users\vagrant\Documents>

```

It seems some kind of permission problem...

**Comment by Sharpie** [comment]
Hmm. That is really unhelpful that `DownloadFile` gives no error message. Was there a non-zero exit code that may indicate what happened?

**Comment by fduranti** [comment]
I'm not sure how to get it ... I've tried with $LASTEXITCODE but it return nothing.
I've tried doing a Try {} Catch statement and it don't get out with any error.
Trying to put a url to a non existant file I get an error so the try catch work correctly.
It seems that the function is able to download the file but not to write it in the documents directory.

This is the code I've used:
```
[System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true}

Try
{	$x=New-Object System.Net.WebClient
	$x.DownloadFile("https://pmaster:8140/packages/current/windows-x86_64/puppet-agent-x64.msi","./puppet-agent-x64.msi")
$x
}
Catch 
      {  $_.Exception  }
```
If i put a non existing file I get the exception: Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."

If the file exist no exception is returned but the file is not in the c:\Users\vagrant\Documents directory. 
Writing in other directory work correctly

**Comment by Sharpie** [comment]
Hmm. Well that's just impolite for DownloadFile to not return an error :)

Are you using a common hypervisor like VirtualBox with a publicly available template? I'm interested in trying to re-produce this.

**Comment by fduranti** [comment]
Yes but it seems some kind of misbehaviour as it only return an error only if it get a http error and not as in this case some other error not related to the http get. 
I’m using virtualbox as hypervisor and a windows box created with template from here without any change https://github.com/boxcutter/windows

**Comment by geoffdavis** [comment]
Hi, I've seen the same problem.

I'm using the _vmware_ provider and the `opentable/win-2012r2-standard-amd64-nocm` image

I ran the same tests that @fduranti was running, and noted the following:
while the installer log was created in `C:\Users\vagrant\Documents`, the download process put the downloaded file in the _parent_ directory, `C:\Users\vagrant`

I'm no PowerShell expert, but my wild guess is that the Powershell script for the installer is using relative paths and should instead be using absolute paths in order to avoid problems with "./" resolving to different things depending on PowerShell object context. The System-Net-Webclient object probably gets initialized with a working directory of `%HOMEDRIVE%%HOMEPATH%`

**Comment by Sharpie** [comment]
@geoffdavis Ah, that is very useful info! I think I may have seen an occurrence of this a few weeks ago, but forced my way through it as I had other yaks to shave that day.

The downloader is likely using `./` as the download location as that is the default for the `stage_installer` capability:

https://github.com/oscar-stack/vagrant-pe_build/blob/0.18.0/lib/pe_build/cap/stage_installer/windows.rb#L17-L31

If anyone wants to take a crack at fixing this, I'd happily review a PR. Otherwise it will sit until I get a quiet moment to break out the PowerShell reference and figure out what environment variable to use.

**Comment by Sharpie** [comment]
This should be fixed in 0.18.1 by using `Get-Item` to fully expand the path to the download location before passing it off to the `System.Net.WebClient` instance.
